internal/services/tenant_settings/resource_tenant_settings.go

```diff
@@ -554,7 +554,7 @@ func (r *TenantSettingsResource) Delete(ctx context.Context, req resource.Delete
 	// restore to previous state
 	previousBytes, diag := req.Private.GetKey(ctx, "original_settings")
 	if diag.HasError() {
-		diag.Append(diag...)
```

Comment:
[nitpick] Consider renaming the variable diag to diags to align with the plural form used elsewhere and improve consistency.